### ISSUE TITLE
Add all example models to `Models`

### ISSUE THREAD

**Issue by Datseris** [JuliaDynamics/Agents.jl#258]
PR #257 introduced the `Models` submodule, that allows for convenient loading of models defined in the examples. 

At the moment, not all example models are in this module, but it is straightforward to add them!

- [x] SIR model for the spread of COVID-19
- [x] Wealth distribution model
- [x] Wright-Fisher model of evolution
- [x] HK (Hegselmann and Krause) opinion dynamics model
- [x] Daisyworld
- [x] Predator-prey dynamics (Wolf, Sheep, Grass)
- [x] Bacterial Growth

**Comment by fbanning** [comment]
Am currently working on this.

**Comment by fbanning** [comment]
Almost done. Will open a PR once I've finished it.

I have one question about growing_bacteria.jl though. Why does the `SimpleCell` function have to be inside the struct?

``` julia
mutable struct SimpleCell <: AbstractAgent
    id::Int
    pos::NTuple{2,Float64}
    length::Float64
    orientation::Float64
    growthprog::Float64
    growthrate::Float64

    ## node positions/forces
    p1::NTuple{2,Float64}
    p2::NTuple{2,Float64}
    f1::NTuple{2,Float64}
    f2::NTuple{2,Float64}

    function SimpleCell(id, pos, l, φ, g, γ)
        a = new(id, pos, l, φ, g, γ, (0.0, 0.0), (0.0, 0.0), (0.0, 0.0), (0.0, 0.0))
        update_nodes!(a)
        return a
    end
end
```

**Comment by Libbum** [comment]
It's a constructor for `SimpleCell` that updates the position value in a custom way (the cell needs some redundant information which assists the mitosis). If that didn't exist, you couldn't do:

```julia
add_agent!((5.0, 5.0), model, 0.0, 0.3, 0.0, 0.1)
add_agent!((6.0, 5.0), model, 0.0, 0.0, 0.0, 0.1)
```
then run the model and expect it to work properly.

Is it causing an issue for the example library? If so, feel free to push up the PR without this working, and we can discuss what a work around could be.

**Comment by fbanning** [comment]
> Is it causing an issue for the example library?

Nope, all good. I was just curious because this kind of structure isn't explained in the tutorial.
Maybe it's worth the time to add an explainer here about why this was written like that.

In other news: I just submitted the PR, see #266 .

**Comment by Datseris** [comment]
Actually here there is no reason to to have this `SimpleCell` function inside struct. And in fact for clarity it is better to put it outside the struct.

**Comment by fbanning** [comment]
> Actually here there is no reason to to have this `SimpleCell` function inside struct. And in fact for clarity it is better to put it outside the struct.

That was what I was wondering about when I was asking "Why does the SimpleCell function have to be **inside** the struct?".
I will move the function out of the struct on monday when I fix the other things mentioned in the PR. 👍

**Comment by Libbum** [comment]
As expected though, this has broken the docs. @Datseris: better to use `add_agent!(agent::AbstractAgent ...)` here or something else?

**Comment by Datseris** [comment]
I am confused, why has this broken the docs...? Having this function in or out is syntactically equivalent.

OH WAIT! @fbanning if you put the function outside you cannot use `new` anymore, you have to replace it with `SimpleCell`.

**Comment by fbanning** [comment]
> you cannot use new anymore, you have to replace it with SimpleCell

Will do tomorrow :)

**Comment by fbanning** [comment]
Should hopefully work now...

**Comment by Libbum** [comment]
Closed via #266
